src/soap/handlers/GetSiteInfo.js

```diff
@@ -14,115 +14,217 @@ import {
   seriesCatalogStart,
   seriesCatalogEnd,
   valueCount,
-  variableTimeInterval
+  variableTimeInterval,
+  qualityControlLevelInfo
 } from '../serializers/series.js'
 import {
+  variableStart,
+  variableInfoType,
+  variableEnd
+} from '../serializers/variable.js'
+import {
+  getSiteInfoResultStart,
   siteStart,
   siteEnd,
   siteInfoStart,
   siteInfoEnd,
   siteInfoType,
   sitesResponseStart,
-  sitesResponseEnd
+  sitesResponseEnd,
+  getSiteInfoResultEnd
 } from '../serializers/site.js'
 import {
   responseStart,
   soapBodyStart,
   soapBodyEnd,
   soapEnvelopeStart,
-  soapEnvelopeEnd
+  soapEnvelopeEnd,
+  soapHeaderStart,
+  soapWsaAction,
+  soapWsaMessageID,
+  soapWsaRelatesTo,
+  soapWsaTo,
+  soapWsuInfo,
+  soapWsseSecurityStart,
+  soapWsseSecurityEnd,
+  soapWsuTimestampStart,
+  soapWsuTimestampEnd,
+  soapHeaderEnd
 } from '../serializers/common.js'
-import {
-  variableStart,
-  variableEnd,
-  variableInfoType
-} from '../serializers/variable.js'
 
-async function* getSiteInfo(_request, { helpers, method, parameters }) {
+export async function* getSiteInfo(
+  request,
+  { date = new Date(), helpers, method, parameters, uniqueid }
+) {
   const { site } = parameters
-  const stationId = site.split(':')[1]
-
-  if (!stationId) throw new Error(`Invalid site parameter: ${site}`)
-
-  // Fetch station and org
-  const station = await helpers.findOne('stations', stationId, {
-    is_enabled: true,
-    is_hidden: false
-  })
-  const organization = await helpers.findOneCached(
-    'organizations',
-    station.organization_id
+  const parts = site.split(':')
+  const org =
+    typeof request.params.org === 'string'
+      ? helpers.org(request.params.org)
+      : undefined
+
+  // Fetch organization
+  const organization = org
+    ? await helpers.findOneCached('organizations', '', {
+        slug: helpers.safeName(org)
+      })
+    : undefined
+
+  // Fetch stations
+  const stations = await helpers.findOne(
```

Comment:
This is confusing. findOne should returns one document (hence the name - find one). Also, this is assigned to a var that is plural (stations). I would suggest findMany. Anything plural is "many" and anything singular is "one".